csharp/src/Ice/ConnectionI.cs

```diff
@@ -2980,7 +2980,10 @@ internal void completed(LocalException ex)
     private int _readStreamPos;
     private int _writeStreamPos;
 
-    // The number of user calls currently executed by the thread-pool (servant dispatch, invocation response, etc.).
+    // An upcall corresponds to application code that the connection executes in a thread of its associated thread pool.
+    // While most upcalls complete synchronously, dispatches are upcalls that can complete asynchronously in another
+    // thread (typically not an Ice thread pool thread). Upcalls include dispatches, AMI (response) continuations,
+    // sent callbacks, and close callbacks.
```

Comment:
Looking at the code, I don't see how close callbacks contribute to the upcall count.

Also, it seems that multiple sent callbacks can count as a single upcall, which I guess is fine.

Reply:
Ok, I was mistaken, the close callback doesn't contribute to the upcall count.